src/tests/modify.test.js

```diff
@@ -0,0 +1,661 @@
+      type: 'integer',
+    },
+    has_premium: {
+      title: 'Premium',
+      type: 'string',
+    },
+    premium_id: {
+      title: 'Premium ID',
+      type: 'boolean',
+    },
+  },
+  'x-jsf-order': ['age', 'quantity', 'has_premium', 'premium_id'],
+  allOf: [
+    {
+      if: {
+        properties: {
+          has_premium: {
+            const: 'yes',
+          },
+        },
+        required: ['has_premium'],
+      },
+      then: {
+        required: ['premium_id'],
+      },
+      else: {
+        properties: {
+          premium_id: false,
+        },
+      },
+    },
+  ],
+};
+
+describe('modify() - structures', () => {
+  // [*6]
+  it('pick fields - basic', () => {
+    const result = modify(schemaTickets, {
+      pick: {
+        fields: ['quantity'],
+      },
+    });
+
+    console.log('xx', result);
+    // Note how the other fields got discarded,
+    // as well the order and allOf got reduced.
+    expect(result.properties).toEqual({
+      quantity: {
+        title: 'Quantity',
+        type: 'integer',
+      },
+    });
+    expect(result.properties.age).toBeUndefined();
+    expect(result.properties.has_premium).toBeUndefined();
+    expect(result.properties.premium_id).toBeUndefined();
+
+    expect(result['x-jsf-order']).toEqual(['quantity']);
+    expect(result.allOf).toEqual([]); // conditional got removed.
+  });
+
+  it.todo('pick fields - keep related conditionals');
+
+  it.todo('pick fields - nested fieldsets');
+
+  it.skip('pick fields - with conditionals', () => {
+    const result = modify(schemaTickets, {
+      pick: {
+        fields: ['has_premium'],
+        onWarn: (ex) => {
+          // A field depends on this one, which was automatically added.
+          // "premium_id" added as it depends on "has_premium".
+          console.warn('Pick was adjusted', ex.message);
+        },
+      },
+    });
+
+    expect(result).toMatchObject({
+      properties: {
+        has_premium: {
+          /*...*/
+        },
+        premium_id: {
+          /*...*/
+        },
+      },
+      allOf: [
+        // the same as schemaTickets
+      ],
+    });
+  });
+
+  // Q:❓ [*7] Would this be really needed?
+  it.skip('omit fields - basic usage', () => {
+    const result = modify(schemaTickets, {
+      omit: 'has_premium',
+    });
+
+    // Note how both "has_premium" and the respective conditional got removed.
+    expect(result).toMatchObject({
+      properties: {
+        age: {
+          /* ... */
+        },
+        quantity: {
+          /* ... */
+        },
+      },
+      'x-jsf-order': ['age', 'quantity'],
+      allOf: [],
+    });
+  });
+
+  // [*8]
+  it.skip('split fields - basic', () => {
+    const result = modify(imagineSomeBasicSchema, {
+      split: {
+        // 💡 Note how "*" is mandatory to ensure
+        // remaining fields has a fallback place.
+        // otherwise the modify() will throw.
+        parts: [
+          ['field_a', 'field_b'],
```

Comment:
What would happen if we picked a field that depend on other fields validations or conditions?

Reply:
That's one of the test cases that we'll take care of `split fields - handling condional fields - disconneted fields`